woodwork/accessor_utils.py

```diff
@@ -147,3 +150,19 @@ def _is_koalas_series(data):
     if ks and isinstance(data, ks.Series):
         return True
     return False
+
+
+def _check_schema_init(data_structure):
+    '''Decorator for Woodwork ColumnAccessor and WoodworkTableAccessor that
+       checks schema initialization
+    '''
+    def inner_decorator(method):
+        @wraps(method)
+        def wrapper(self, *args, **kwargs):
+            if self._schema is None:
+                msg = (f"Woodwork not initialized for this {data_structure}. "
+                       f"Initialize by calling {data_structure}.ww.init")
+                raise WoodworkNotInitError(msg)
+            return method(self, *args, **kwargs)
+        return wrapper
+    return inner_decorator
```

Comment:
Seems like this would also work here.  Then you don't need to include the explicit class name in each decorator call.
```suggestion
def _check_schema_init(method):
    '''Decorator for WoodworkColumnAccessor and WoodworkTableAccessor that
       checks schema initialization
    '''
    @wraps(method)
    def wrapper(self, *args, **kwargs):
        class_name = type(self).__name__
        if self._schema is None:
            msg = (f"Woodwork not initialized for this {class_name}. "
                   f"Initialize by calling {class_name}.ww.init")
            raise WoodworkNotInitError(msg)
        return method(self, *args, **kwargs)
    return wrapper
```

Reply:
Actually, it looks like you couldn't use `type(self).__name__` since `self` is a column or table accessor.  But you could still determine the correct pandas collection type by doing something like `isinstance(self, PandasColumnAccessor)`, etc.  Seems better than putting a class name explicitly into all the decorator calls that could potentially be typoed.

Reply:
And then, for that matter, maybe it would make more sense to do what @tamargrey suggested and include some kind of check for the type of `self`:
```suggestion
def _check_schema_init(method):
    '''Decorator for WoodworkColumnAccessor and WoodworkTableAccessor that
       checks schema initialization
    '''
    @wraps(method)
    def wrapper(self, *args, **kwargs):
        if isinstance(self, WoodworkTableAccessor):
            data_structure = "DataFrame"
        elif isinstance(self, WoodworkColumnAccessor):
            data_structure = "Series"
        else:
            raise TypeError("decorator was applied to a non-accessor method")

        if self._schema is None:
            msg = (f"Woodwork not initialized for this {data_structure}. "
                   f"Initialize by calling {data_structure}.ww.init")
            raise WoodworkNotInitError(msg)

        return method(self, *args, **kwargs)

    return wrapper
```

Reply:
I think we'll run into a circular import issue since this function needs to be defined before the accessor classes in order to be used.  We could import the accessors within the method to avoid the circular import.

Another approach would be to check if Table or Column are in the class name, assuming we're willing to limit our subclasses to always include TableAccessor or ColumnAccessor in the name

Reply:
I'd be okay with importing the accessors within the method if need be. But I would also be okay with keeping these as two separate functions.

My thought process behind that is that in having separate functions we 
1. Keep the decorator function simpler and easier to understand
2. Can more easily add additional DataFrame or Series-specific checks if we decide to expand the checks performed in the decorator (for example we were considering adding a schema validity check as a part of #724, and we'd need separate validity checks for dataframes vs series)

Reply:
I think the local import to avoid circular imports is fine.  They should probably go just inside of the decorator body just before the wrapper function is defined.  Then the overhead is restricted to when any module is parsed and the decorators are executed to yield the wrapper function (one-time cost).

However, I'm also fine with keeping them separate.  Really don't have a specific preference here.

The one thing I would say is that, if we keep the multi-use decorator, let's not look for a substring in the class name.  I often really try to avoid "parsing" class or function names as part of a library's API.  That makes it really likely that, at some point, someone will change the name of some resource without realizing that there's some parsing facility somewhere in the library that expects the names of things to have a certain format.  And that's not something you can grep for.  On the other hand, bare identifiers that refer to class or function objects (or string literals that refer to the same) are greppable and can be updated fairly easily if someone ever renames anything.